Some TU Datasets do not work -  Error when loading

Unfortunately, some TU Datasets can not be loaded. The following example illustrates the problem for dataset "Cuneiform":
```julia
using MLDatasets
tudata = TUDataset("Cuneiform")
```
The following error message occurs:

```julia
ERROR: at row 1, column 1 : ErrorException("file entry \"0,\" cannot be converted to Int64")
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] dlm_fill(T::DataType, offarr::Vector{Vector{Int64}}, dims::Tuple{Int64, Int64}, has_header::Bool, sbuff::String, auto::Bool, eol::Char)
    @ DelimitedFiles /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:515
  [3] readdlm_string(sbuff::String, dlm::Char, T::Type, eol::Char, auto::Bool, optsd::Dict{Symbol, Union{Char, Integer, Tuple{Integer, Integer}}})
    @ DelimitedFiles /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:471
  [4] readdlm_auto(input::String, dlm::Char, T::Type, eol::Char, auto::Bool; opts::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ DelimitedFiles /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:247
  [5] readdlm_auto
    @ /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:233 [inlined]
  [6] #readdlm#6
    @ /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:226 [inlined]
  [7] readdlm
    @ /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:226 [inlined]
  [8] #readdlm#1
    @ /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:57 [inlined]
  [9] readdlm
    @ /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:57 [inlined]
 [10] TUDataset(name::String; dir::Nothing)
    @ MLDatasets ~/.julia/packages/MLDatasets/OYCcg/src/datasets/graphs/tudataset.jl:74
 [11] TUDataset(name::String)
    @ MLDatasets ~/.julia/packages/MLDatasets/OYCcg/src/datasets/graphs/tudataset.jl:54

caused by: file entry "0," cannot be converted to Int64
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:35
  [2] store_cell(dlmstore::DelimitedFiles.DLMStore{Int64}, row::Int64, col::Int64, quoted::Bool, startpos::Int64, endpos::Int64)
    @ DelimitedFiles /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:388
  [3] dlm_fill(T::DataType, offarr::Vector{Vector{Int64}}, dims::Tuple{Int64, Int64}, has_header::Bool, sbuff::String, auto::Bool, eol::Char)
    @ DelimitedFiles /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:510
  [4] readdlm_string(sbuff::String, dlm::Char, T::Type, eol::Char, auto::Bool, optsd::Dict{Symbol, Union{Char, Integer, Tuple{Integer, Integer}}})
    @ DelimitedFiles /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:471
  [5] readdlm_auto(input::String, dlm::Char, T::Type, eol::Char, auto::Bool; opts::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ DelimitedFiles /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:247
  [6] readdlm_auto
    @ /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:233 [inlined]
  [7] #readdlm#6
    @ /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:226 [inlined]
  [8] readdlm
    @ /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:226 [inlined]
  [9] #readdlm#1
    @ /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:57 [inlined]
 [10] readdlm
    @ /usr/share/julia/stdlib/v1.8/DelimitedFiles/src/DelimitedFiles.jl:57 [inlined]
 [11] TUDataset(name::String; dir::Nothing)
    @ MLDatasets ~/.julia/packages/MLDatasets/OYCcg/src/datasets/graphs/tudataset.jl:74
 [12] TUDataset(name::String)
    @ MLDatasets ~/.julia/packages/MLDatasets/OYCcg/src/datasets/graphs/tudataset.jl:54
  
 ```

Other problems arise for Fingerprint:

```julia
tudata = TUDataset("Fingerprint")
```

```julia
AssertionError: all(sort(unique(graph_indicator)) .== 1:length(unique(graph_indicator)))
Stacktrace:
 [1] TUDataset(name::String; dir::Nothing)
   @ MLDatasets ~/.julia/packages/MLDatasets/OYCcg/src/datasets/graphs/tudataset.jl:66
 [2] TUDataset(name::String)
   @ MLDatasets ~/.julia/packages/MLDatasets/OYCcg/src/datasets/graphs/tudataset.jl:54
 [3] top-level scope
   @ REPL[5]:1
   ```

--- a/src/datasets/graphs/tudataset.jl
+++ b/src/datasets/graphs/tudataset.jl
@@ -72,13 +72,13 @@ function TUDataset(name; dir=nothing)
     # LOAD OPTIONAL FILES IF EXIST
     
     node_labels = isfile(joinpath(d, "$(name)_node_labels.txt")) ?
-                    readdlm(joinpath(d, "$(name)_node_labels.txt"), Int) |> vec :
+                    readdlm(joinpath(d, "$(name)_node_labels.txt"), ',', Int) |> vec :
                     nothing
     edge_labels = isfile(joinpath(d, "$(name)_edge_labels.txt")) ?
-                    readdlm(joinpath(d, "$(name)_edge_labels.txt"), Int) |> vec :
+                    readdlm(joinpath(d, "$(name)_edge_labels.txt"), ',', Int) |> vec :
                     nothing
     graph_labels = isfile(joinpath(d, "$(name)_graph_labels.txt")) ?
-                    readdlm(joinpath(d, "$(name)_graph_labels.txt"), Int) |> vec :
+                    readdlm(joinpath(d, "$(name)_graph_labels.txt"), ',', Int) |> vec :
                     nothing
 
     node_attributes = isfile(joinpath(d, "$(name)_node_attributes.txt")) ?

--- a/src/datasets/graphs/tudataset.jl
+++ b/src/datasets/graphs/tudataset.jl
@@ -72,13 +72,13 @@ function TUDataset(name; dir=nothing)
     # LOAD OPTIONAL FILES IF EXIST
     
     node_labels = isfile(joinpath(d, "$(name)_node_labels.txt")) ?
-                    readdlm(joinpath(d, "$(name)_node_labels.txt"), ',', Int) |> vec :
+                    readdlm(joinpath(d, "$(name)_node_labels.txt"), ',', Int)' |> collect |> maybesqueeze :
                     nothing
     edge_labels = isfile(joinpath(d, "$(name)_edge_labels.txt")) ?
-                    readdlm(joinpath(d, "$(name)_edge_labels.txt"), ',', Int) |> vec :
+                    readdlm(joinpath(d, "$(name)_edge_labels.txt"), ',', Int)' |> collect |> maybesqueeze :
                     nothing
     graph_labels = isfile(joinpath(d, "$(name)_graph_labels.txt")) ?
-                    readdlm(joinpath(d, "$(name)_graph_labels.txt"), ',', Int) |> vec :
+                    readdlm(joinpath(d, "$(name)_graph_labels.txt"), ',', Int)' |> collect |> maybesqueeze :
                     nothing
 
     node_attributes = isfile(joinpath(d, "$(name)_node_attributes.txt")) ?

--- a/test/datasets/graphs.jl
+++ b/test/datasets/graphs.jl
@@ -252,3 +252,37 @@ end
         end
     end
 end
+
+@testset "TUDataset - Cuneiform" begin
+    data  = TUDataset("Cuneiform")
+
+    @test data.num_nodes == 5680
+    @test data.num_edges == 23922
+    @test data.num_graphs == 267
+
+    @test data.num_nodes == sum(g->g.num_nodes, data.graphs)
+    @test data.num_edges == sum(g->g.num_edges, data.graphs)
+    @test data.num_edges == sum(g->length(g.edge_index[1]), data.graphs)
+    @test data.num_edges == sum(g->length(g.edge_index[2]), data.graphs)
+    @test data.num_graphs == length(data) == length(data.graphs)
+
+    i = rand(1:length(data))
+    di = data[i]
+    @test di isa NamedTuple
+    g, targets = di.graphs, di.targets
+    @test targets isa Int
+    @test g isa Graph
+    @test all(1 .<= g.edge_index[1] .<= g.num_nodes)
+    @test all(1 .<= g.edge_index[2] .<= g.num_nodes)
+
+    # graph data
+    @test size(data.graph_data.targets) == (data.num_graphs, )
+
+    # node data
+    @test size(g.node_data.features) == (3, g.num_nodes)
+    @test size(g.node_data.targets) == (2, g.num_nodes)
+
+    # edge data
+    @test size(g.edge_data.features) == (2, g.num_edges)
+    @test size(g.edge_data.targets) == (g.num_edges, )
+end

